### DISCUSSION TITLE
Unable to modify a key that is of type `KeyMut`

### DISCUSSION THREAD

**Discussion by johnsonw** [toml-rs/toml#645] [Q&A]
Hello, I am in the process of mutably traversing a TOML file using toml_edit. When I come across a key that I need to change I am not able to change the key's value even though the key is of type `KeyMut`. The key cannot be changed because the property is not public and I can't seem to find any methods that allow me to mutate its value. As an example, let's say that I have the following in my toml file:

```
[host1.network_interface.eth0]
name = "name"
...
```

In this example, I want to change the `eth0` key to `enp0s2`. Is it possible to do this using the `toml_edit` crate?

**Comment by epage** [comment]
If you look at [HashMap](https://doc.rust-lang.org/std/collections/struct.HashMap.html), you can't edit keys in-place.  You have to remove the entry and insert a new one.  The same for `toml_edit`.  The role of `KeyMut` is for changing the aspects of a key that are independent of its lookup-properties.

**Reply by johnsonw** [reply]
Fair enough. Thank you for the clarification 👍
